[Fix] Replaced automatable PICS in SEAR tests

j-ororke opened the pull request:
- Replaced automatable PICS in SEAR 1.3, 1.5, and 1.6 test modules with attributes gathered from endpoint.

### Testing

j-ororke committed:
[Fix] Replaced automatable PICS in SEAR tests:
- Replaced automatable PICS in SEAR 1.3, 1.5, and 1.6 test modules.

j-ororke committed:
Updated TC_SEAR_1_6 test module:
- resolved linting issue

restyled-commits committed:
Restyled by autopep8

j-ororke committed:
Merge branch 'master' into remove_PICS_from_SEAR_tests

j-ororke committed:
Merge branch 'master' into remove_PICS_from_SEAR_tests

j-ororke committed:
Updated TC_SEAR_1_5 test module:
- Replaced C02.Rsp with SkipArea command id check against accepted commands list for SearchArea cluster

restyled-commits committed:
Restyled by autopep8

j-ororke committed:
Merge branch 'master' into remove_PICS_from_SEAR_tests

ReneJosefsen commented:
Does this align with the Test Plan? I think this is a good addition but it needs to be aligned with the Test Plan and given the timing, this might be something we need to postpone to 1.5?

cecille commented on src/python_testing/TC_SEAR_1_6.py:
```diff
@@ -72,7 +72,7 @@ async def read_progress(self, step):
 
     def TC_SEAR_1_6(self) -> list[str]:
-        return ["SEAR.S", "SEAR.S.A0005", "SEAR.S.A0000", "SEAR.S.A0002", "SEAR.S.M.HAS_MANUAL_OPERATING_STATE_CONTROL"]
+        return ["SEAR.S", "SEAR.S.M.HAS_MANUAL_OPERATING_STATE_CONTROL"]
```
Actually, let's just leave this test as-is for now. The top level pics we can deal with separately and this will mess up the TH

j-ororke replied:
Thank you,
This has been updated to leave the top level PICS as-is and should now be resolved.

cecille commented on src/python_testing/TC_SEAR_1_5.py:
```diff
@@ -89,7 +89,7 @@ async def send_cmd_skip_area_expect_response(self, step, skipped_area, expected_
 
     def TC_SEAR_1_5(self) -> list[str]:
-        return ["SEAR.S", "SEAR.S.C02.Rsp"]
+        return ["SEAR.S"]
```
Same here - let's leave the top level PICS as-is for now and we'll deal with the top level PICS separately.

j-ororke replied:
This has been updated to leave the top level PICS as-is.

j-ororke committed:
Update TC_SEAR_1_5.py

Updating to leave top level PIC's unchanged

j-ororke committed:
Update TC_SEAR_1_6.py

Updating to leave top level PICS alone

j-ororke committed:
Merge branch 'master' into remove_PICS_from_SEAR_tests

restyled-commits committed:
Restyled by autopep8

j-ororke committed:
Merge branch 'master' into remove_PICS_from_SEAR_tests

j-ororke committed:
Merge branch 'master' into remove_PICS_from_SEAR_tests

j-ororke committed:
Updating TC_SEAR_1_3, TC_SEAR_1_5, and TC_SEAR_1_6 test modules:
- Updating to using attribute_guard, command_guard, and feature_guard functions where applicable.

cecille commented on src/python_testing/TC_SEAR_1_6.py:
```diff
@@ -94,6 +94,17 @@ async def test_TC_SEAR_1_6(self):
 
+        attributes = Clusters.ServiceArea.Attributes
+
+        if not await self.attribute_guard(endpoint=self.endpoint, attribute=attributes.SupportedAreas):
```
Given that we now have the top-level guards, this should be changed over to use those.

Also, I'm pretty sure the pics_ function in this test is incorrect - it looks like the function on line 83 is suppose to be pics_TC_SEAR_1_6, which is probably why these were added - the pics definition seems wrong.

j-ororke replied:
I have changed the function name for the pics to be pics_TC_SEAR_1_* based on the test being run so that the pics can be found during testing if they are needed.

j-ororke committed:
Updating TC_SEAR_1_* test modules:
- Changing to using correct names for pics function to be found by test runner script.

j-ororke committed:
Merge branch 'master' into remove_PICS_from_SEAR_tests
